### PR TITLE
Add MiMo-VL conversion validation utilities

### PR DESCRIPTION
**Pull request by yqf66**
#### Before submitting

- [x] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [x] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

## PR Type

Model checkpoint adaptation / conversion support.

## Description

This PR adds conversion and validation utilities for `MiMo-VL-7B-SFT`.

`MiMo-VL-7B-SFT` does not introduce a new model architecture in PaddleFormers. Its checkpoint follows the Qwen2.5-VL architecture and can be loaded by the existing `Qwen2_5_VLForConditionalGeneration` implementation. Therefore this PR does not add a new `paddleformers/transformers/mimo_vl` modeling directory.

Instead, this PR focuses on:

- Converting the HuggingFace safetensors checkpoint to a Paddle-compatible checkpoint.
- Verifying that the converted Paddle checkpoint can be loaded by `Qwen2_5_VLForConditionalGeneration`.
- Comparing text-only logits between the original HF checkpoint and the converted Paddle checkpoint.
- Providing usage notes for publishing the converted checkpoint to the Paddle model hub.

## Why No New Model Files Are Added

MiMo-VL reuses the Qwen2.5-VL model structure. Duplicating the Qwen2.5-VL implementation under a separate `mimo_vl` module would add maintenance cost without introducing new model behavior.

The converted checkpoint should be loaded through:

```python
from paddleformers.transformers import Qwen2_5_VLForConditionalGeneration

model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
    "<mimo-vl-paddle-checkpoint>",
    convert_from_hf=True,
    use_safetensors=True,
    load_checkpoint_format="legacy",
    dtype="bfloat16",
)
```

### 1. Paddle Checkpoint Load

Command:

```bash
CUDA_VISIBLE_DEVICES=7 LOAD_STATE_DICT_THREAD_NUM=8 python - <<'PY'
import sys, paddle
paddle.set_device("gpu:0")
sys.path.insert(0, "/path/to/PaddleFormers")
from paddleformers.transformers import Qwen2_5_VLForConditionalGeneration

Qwen2_5_VLForConditionalGeneration.from_pretrained(
    "<mimo-vl-paddle-checkpoint>",
    convert_from_hf=True,
    use_safetensors=True,
    load_checkpoint_format="legacy",
    low_cpu_mem_usage=True,
    dtype="bfloat16",
)
print("verify_load_pass")
PY
```

Result:

```text
All model checkpoint weights were used when initializing Qwen2_5_VLForConditionalGeneration.
verify_load_pass
```

### 2. Text-Only Smoke Inference

Result:

```text
smoke_pass=True
gen_time_sec=1.682
```

### 3. HF vs Paddle Logits Consistency

The same text input was tokenized once and fed to both the original HF checkpoint and the converted Paddle checkpoint.

Result:

```json
{
  "shape": [1, 8, 151680],
  "max_abs_diff": 0.0005908012390136719,
  "mean_abs_diff": 2.8572936571436003e-05,
  "p99_abs_diff": 0.000240325927734375,
  "allclose": true,
  "atol": 0.001,
  "rtol": 0.001,
  "hf_top1_token_id": 56177,
  "paddle_top1_token_id": 56177,
  "top1_match": true
}
```

Conclusion:

The converted Paddle checkpoint is numerically aligned with the original HF checkpoint for text-only forward logits under `atol=1e-3, rtol=1e-3`.

## Notes

- The converted checkpoint is saved as safetensors.
- `load_checkpoint_format="legacy"` is used for stable local loading. The flex checkpoint path may trigger memory pressure or AOA-related loading issues for this checkpoint.
- The checkpoint keys are HF-style after saving, so `convert_from_hf=True` is required when loading the converted Paddle checkpoint.

## Checklist

- Add MiMo-VL conversion utility.
- Add HF vs Paddle logits verification utility.
- Verify converted Paddle checkpoint can be loaded.
- Run text-only smoke inference.
- Run HF vs Paddle logits consistency check.
- Upload converted checkpoint to Paddle model hub.

### CODE REVIEW

**Comment by yqf66** [comment]
您好，我想说明一下这个 PR 中 MiMo-VL 模型的特殊情况。

MiMo-VL 本身并不是一个需要在 PaddleFormers 中新增独立模型结构的模型。根据官方实现和模型说明，MiMo-VL 与 Qwen2.5-VL 架构兼容，因此本 PR 并没有新增独立的 `modeling.py`、`configuration.py`、tokenizer、processor 或新的 `paddleformers/transformers/mimo_vl` 模型目录。

这个 PR 的主要目标是提供 MiMo-VL checkpoint 到 PaddleFormers 现有 Qwen2.5-VL 实现的转换与验证工具。也就是说，这个 PR 更接近于“兼容架构下的权重转换与验证支持”，而不是一个标准意义上的“新增模型架构”PR。

因此，部分默认针对新增模型的 CI/CE 检查可能并不完全适用于这个 PR。例如，如果 CE 要求提交新的模型文件、模型配置类或模型专属单测，这里实际上没有对应的新增模型实现，因为 MiMo-VL 的推理逻辑是复用 PaddleFormers 中已有的 Qwen2.5-VL 实现。

希望维护者给出建议，指导该模型如何做合并，如果目前将脚本放在 `tests/integration_test` 目录不太合适，我也可以根据维护者建议，将转换和验证脚本移动到更合适的位置
